lib/evmone/baseline.cpp

```diff
@@ -72,10 +72,10 @@ inline evmc_status_code check_requirements(
 
 
 /// Implementation of a generic instruction "case".
-#define DISPATCH_CASE(OPCODE)                                          \
-    case OPCODE:                                                       \
-        if (code_it = invoke(op2fn::OPCODE, state, code_it); !code_it) \
-            goto exit;                                                 \
+#define DISPATCH_CASE(OPCODE)                                   \
+    case OPCODE:                                                \
+        if (code_it = invoke<OPCODE>(state, code_it); !code_it) \
```

Comment:
Why is helper required, can't it be
```suggestion
        if (code_it = invoke(instr::impl<OPCODE>, state, code_it); !code_it) \
```

Reply:
It can be. This comes from the way I will use it later.